### PR TITLE
Mccalluc/datacite collections

### PR DESCRIPTION
**Pull request by mccalluc**
This is in the mockup in #1812, but I don't see an issue.
- @tsliaw -- Review design.
- @john-conroy -- Code review. Not sure I've organized this well.

### CODE REVIEW

File: context/app/static/js/components/Detail/Citation/style.js [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+
+const FlexRight = styled.div`
+  display: flex;
+  margin-left: auto;
+`;
+
+const StyledOpenInNewRoundedIcon = styled(OpenInNewRoundedIcon)`
+  font-size: 1.1rem;
+  vertical-align: text-bottom;
+  margin-left: 0.2rem;
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Citation/style.js:22]:
```suggestion
  margin-left: ${(props) => props.theme.spacing(0.5)}px;
```

File: context/app/static/js/components/Detail/Citation/Citation.jsx [hunk 2]
```diff
@@ -15,12 +16,25 @@ function buildNLMCitation({ contributors, title, timestamp }) {
...
-      <SectionHeader>Citation</SectionHeader>
+      <Flex>
+        <SectionHeader>Citation</SectionHeader>
+        <FlexRight>
+          <Button
+            color="primary"
+            variant="contained"
+            component="a"
+            target="_blank"
+            href={`https://search.datacite.org/works/${doi}`}
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Citation/Citation.jsx:32]: We shouldn't need to encode the doi, right?

**Review by john-conroy — COMMENTED** [review]
I would lean towards creating a reusable button component `OutboundLinkButton` or similar which leverages the existing `OutboundLink` component.

If we think this will only be used by collections I'd put it in the collections dir in components (where other components that are in detail should probably be).

**Review by tsliaw — APPROVED** [review]

File: context/app/static/js/components/Detail/Citation/Citation.jsx (hunk 2 above)
Comment (mccalluc) [reply context/app/static/js/components/Detail/Citation/Citation.jsx:32]: I don't think so: If the service were structured differently, we might, but this works with a string like `10.35079/hbm876.xnrh.336` right in the path URL: We don't want to encode the `/` or anything like that.

**Comment by mccalluc** [comment]
> I would lean towards creating a reusable button component `OutboundLinkButton` or similar which leverages the existing `OutboundLink` component.

Done, sort-of: Just putting `OutboundLink` inside the Button doesn't work because then you can only click exactly on the text, not anywhere else in the button. Instead, I've factored out the google analytics callback, so it can be called onClick, either form the original Link, or the new Button.

**Pull request merged**